src/realm/object-store/sync/sync_session.cpp

```diff
@@ -1267,7 +1267,12 @@ void SyncSession::wait_for_download_completion(util::UniqueFunction<void(Status)
 uint64_t SyncSession::register_progress_notifier(std::function<ProgressNotifierCallback>&& notifier,
                                                  ProgressDirection direction, bool is_streaming)
 {
-    return m_progress_notifier.register_callback(std::move(notifier), direction, is_streaming);
+    int64_t pending_query_version = 0;
+    assert_mutex_unlocked();
```

Comment:
`register_progress_notifier()` needs to be annotated as `REQUIRES(!m_state_mutex)` rather than this.